rsocket/framing/FrameTransport.cpp

```diff
@@ -161,44 +160,25 @@ void FrameTransport::onError(std::exception_ptr eptr) {
 }
 
 void FrameTransport::request(int64_t n) {
-  Lock lock(mutex_);
-
-  if (!connection_) {
-    // request(n) can be delivered during disconnecting.  We don't care for it
-    // anymore.
-    return;
-  }
-
-  if (writeAllowance_.release(n) > 0) {
-    // There are no pending wfrites or we already have this method on the
-    // stack.
-    return;
-  }
-
-  drainWrites(lock);
+  // we are expecting we can write output without back pressure
+  CHECK_EQ(n, std::numeric_limits<int64_t>::max());
 }
 
 void FrameTransport::cancel() {
   VLOG(6) << "cancel";
   terminateProcessor(folly::exception_wrapper());
 }
 
-void FrameTransport::outputFrameOrEnqueue(std::unique_ptr<folly::IOBuf> frame) {
+void FrameTransport::outputFrameOrDrop(std::unique_ptr<folly::IOBuf> frame) {
   Lock lock(mutex_);
```

Comment:
The lock is not needed now at all (in the whole class). We don't support resumption from multiple threads now so we never access the member variables from a different thread.
In this diff I don't want to remove more than its necessary.